### ISSUE TITLE
After adding mlx.nn.BatchNorm, evaluation results differ in train.py vs. test.py

### ISSUE THREAD

**Issue by menzHSE** [menzHSE/mlx-cifar-10-cnn#5]
After adding mlx.nn.BatchNorm, evaluation results differ in train.py vs. test.py. 
Model seems to be correctly set to ```eval()``` and ```train()``` 

```
Epoch   27: Loss 1.07467 | Train accuracy  0.785 | Test accuracy  0.542 | Throughput     688.19 images/second |  Time   75.054 (s)
Epoch   28: Loss 1.05003 | Train accuracy  0.790 | Test accuracy  0.541 | Throughput     688.58 images/second |  Time   74.930 (s)
Epoch   29: Loss 1.03433 | Train accuracy  0.799 | **Test accuracy  0.543** | Throughput     706.21 images/second |  Time   73.218 (s)
```

```
$ python test.py  --model models/model_CIFAR-100_029.npz --dataset=CIFAR-100
Loaded model for CIFAR-100 from models/model_CIFAR-100_029.npz
Starting testing ...
....
Test accuracy: 0.5409345030784607
```
